pkg/addon/create.go

```diff
@@ -247,7 +247,7 @@ func writeRequiredFiles(addonPath string, tmplContent, readmeContent, metadataCo
 	templateFilePath := path.Join(addonPath, TemplateFileName)
 	err := os.WriteFile(templateFilePath,
 		tmplContent,
-		0644)
+		0600)
```

Comment:
This is a template for the user. We want other user to be able to access it, so please keep `0644`. You can ignore this by `// nolint:gosec`

Reply:
Alright, thank you.

Reply:
I've fixed this.